serverlessworkflow/modules/ROOT/pages/data-index/data-index-service.adoc

```diff
@@ -0,0 +1,225 @@
+:dev_services_url: https://quarkus.io/guides/dev-services
+:test_containers_url: https://www.testcontainers.org/
+// Referenced documentation pages
+//Common constants
+:data_index_ref: Data Index
+:workflow_instance: process instance
+:workflow_instances: {workflow_instance}s
+
+[#data-index-service]
+== {data_index_ref} service deployment.
+
+{data_index_ref} service can be deployed referencing directly a distributed {data_index_ref} image. There are different images provided that take into account what persistence layer is required in each case.
+In each distribution, there are some properties to configure things like the connection with the database or the communication with other services.
+The goal is to configure the container to allow to process ProcessInstances and Jobs *events* that incorporate their related data, to index and store that in the database and finally, to provide the xref:data-index/data-index-core-concepts.adoc#data-index-graphql[{data_index_ref} GraphQL] endpoint to consume it.
+
+[#data-index-service-distributions]
+=== {data_index_ref} distributions
+{data_index_ref} service can be deployed referencing directly a distributed {data_index_ref} image. Here there are the different {data_index_ref} image distributions that can be found in link:{distributions_url}[Quay.io/kiegroup]:
+
+image::data-index/data-index-distributions.png[Image of data-index different available distributions]
+
+=== {data_index_ref} service deployment
+
+When the {data_index_ref} service is deployed, a `configMap` resource is created for the `application.properties` configuration of the {data_index_ref} service.
+
+The name of the `configMap` resource consists of the name of the {data_index_ref} service and the suffix `-properties`, as shown in the following example:
+
+.Example `configMap` resource generated during {data_index_ref} service deployment
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: data-index-properties
+data:
+  application.properties : |-
+    property1=value1
+    property2=value2
+----
+
+The `application.properties` data of the `configMap` resource is mounted in a volume to the container of the {data_index_ref} service. Any runtime properties that you add to the `application.properties` section override the default application configuration properties of the {data_index_ref} service.
+
+When the `application.properties` data of the `configMap` is changed, a rolling update modifies the deployment and configuration of the {data_index_ref} service.
+
+---
+
+=== {data_index_ref} service using PostgreSQL persistence
+
+{data_index_ref} service does not initialize its database schema automatically. To initialize the database schema, you need to enable Flyway migration by setting QUARKUS_FLYWAY_MIGRATE_AT_START=true.
+
+.Example `DataIndex` resource with PostgreSQL persistence enabled:
+[source,yaml]
+----
+apiVersion: app.kiegroup.org/v1beta1
+kind: KogitoSupportingService
```

Comment:
`KogitoSupportingService` was with the `kogito-operator`. I am not sure we want that in the documentation
cc @ricardozanini

Reply:
we do not

Reply:
@radtriste @ricardozanini  I replaced the example that we showed in kie-docs by the one that is working in timeouts example

Reply:
Good with the change